sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -0,0 +1,644 @@
+
+## Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification section 6.3.3 "Feature test macros".  Therefore, an
+implementation supporting this extension must predefine the macro
+`SYCL_EXT_ONEAPI_MATRIX` to one of the values defined in the table below.
+Applications can test for the existence of this macro to determine if the
+implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's APIs the implementation supports.
+
+[frame="none",options="header"]
+|======================
+|Value |Description
+|1     |Initial extension JIT implementation on Intel AMX and DPAS. load, store, mad, fill, piece-wise operations, and the query interface are supported 
+|2     |joint matrix type has a new `use` parameter. `layout` on matrix is unused. JIT implementation on Intel AMX and DPAS. load, store, mad, fill, piece-wise operations, and the query interface are supported 
+|======================
+
+## New `joint_matrix` class
+We introduce a new class called `joint_matrix`. The user needs to specify the type of the elements, shape, the matrix use, the memory layout, and the memory scope of the matrix. This results in the following description:
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+template <typename T, use Use,
+          size_t Rows=sycl::dynamic_extent, size_t Cols=sycl::dynamic_extent,
+          layout Layout = layout::unused, typename Group = sub_group>
+struct joint_matrix {
+    joint_matrix(Group g) {}
+};
+}
+```
+
+#### Shape
+The same class, `joint_matrix`, should handle both cases where sizes are constant (GPU case) and when sizes are variable (CPU case). Note that a Intel AMX 2d tile register permits sizes up to 1024 (16rowsx64cols) bytes. The ability to define only one interface for both makes it possible to give the user a way to make use of the flexibility introduced by the CPU but at the same time save resources on the GPU. We use `sycl::dynamic_extent`  to differentiate between static and dynamic sizes.
+
+IMPORTANT: In the current implementation, only the static extent is supported
+
+#### Use
+Specifying the usage of the matrix: matrix left (A), matrix right (B) or accumulator +(C)+ is required by backend implementations to reason about the layout of the matrix in registers.
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+enum class use {
+  a,
+  b,
+  accumulator
+};
+}
+```
+
+IMPORTANT: In both AMX and DPAS support, the `use` template parameter is required
+
+
+#### Layout
+Besides row major and column major layouts, `layout` is flexible enough to introduce custom layouts such as packed layout.
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+enum class layout {
+  row_major,
+  col_major,
+  packed,
+  unused
+};
+}
+```
+
+IMPORTANT: In both AMX and DPAS support, the `layout` template parameter is unused and will be ignored if specified
```

Comment:
If this document will focus on the matrix extension interfaces apparent to the user for the Intel backend only, you could consider completely omitting this "Important" statement, and describe joint_matrix as above without the template parameter as you did with:
```
joint_matrix<int8_t, use::a, tM, tN> tA(sg);
```

However admittedly this could be perhaps considered a little dodgy since the user still could construct a joint_matrix with the layout template parameter so long as it is layout::unused, assuming that we keep it as it is.
```